providers/terraform-provider-csbsqlserver/connector/connector.go

```diff
@@ -81,6 +90,24 @@ func (c *Connector) withConnection(callback func(db *sql.DB) error) error {
 	return callback(db)
 }
 
+func (c *Connector) withTransaction(callback func(db databaseActor) error) error {
```

Comment:
**suggestion**: naming both closure arguments `db` may cause confusion. I think we can safely define the `callback` argument as `func(databaseActor) error`.